lib/pavilion/test_run.py

```diff
@@ -412,15 +412,26 @@ def _save_config(self):
 
         config_path = self.path/'config'
 
+        # make lock
+        lock_path = self.path/'config.lockfile'
+        config_lock = lockfile.LockFile(
+            lock_path,
+            group=self._pav_cfg.shared_group
+        )
+
```

Comment:
There's a much easier way:
https://rcrowley.org/2010/01/06/things-unix-can-do-atomically.html

Reply:
Nope, I'm wrong. If you were just creating the file, not replacing it, writing it to a tmp file and `os.rename`ing it is atomic. But since you have to delete the original first, it isn't. So you do have to use a lock.